src/smtsolvers/CoreSMTSolver.cc

```diff
@@ -934,6 +934,26 @@ void CoreSMTSolver::finalizeResolutionProof(CRef finalConflict) {
     resolutionProof->endChain(CRef_Undef);
 }
 
+
+CRef CoreSMTSolver::logUnitClauseDerivationAtLevelZero(CRef cref) {
+    assert(logsResolutionProof() and decisionLevel() == 0);
+    resolutionProof->beginChain(cref);
+    Clause const & c = ca[cref];
+    assert(c.size() >= 2);
+    assert(value(c[0]) == l_Undef);
+    for (unsigned k = 1; k < c.size(); ++k) {
+        Lit l = c[k];
+        assert(value(l) == l_False);
+        Var v = var(l);
+        assert(level(v) == 0);
+        assert(reason(v) != CRef_Fake and reason(v) != CRef_Undef);
```

Comment:
In case of assertion failure, I believe it is better to split the checks to know exactly what happened.

Reply:
Changed!